scripts/pre_commit_linter.py

```diff
@@ -714,11 +718,142 @@ def _check_spacing(all_files):
     return summary_messages
 
 
+def _check_comments(all_files):
+    """This functions checks for a period at the end of comment."""
+    print 'Starting comment checks'
+    print '----------------------------------------'
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.py')]
+    message = 'There should be a period at the end of the comment.'
+    failed = False
+    for filename in files_to_check:
+        f = open(filename, 'r')
+        content = f.readlines()
+        length = len(content)
+        line_num = 0
+        while line_num < length:
```

Comment:
You could do a for loop: `for line_num  in range(file_length)`, which would help you do a lot of book keeping

Reply:
If I do that, the variable ``line_num`` will not be updated according to the content inside the loop.

Reply:
Ah, the strategy that I had in mind was roughly like this (I'm being very abstract):
```
for i in range(file_length):
    strip spaces from lines
    if line is a comment:
        if line is the last line or next line is not a comment:
             check that correct punctuation ends comment
 ```

Reply:
But we need to update i within the loop as well as in case of multiline comments and that cannot be done in this case.

Reply:
No need, because once the condition in the checker is not satisfied, it'll just loop to the next line.

Reply:
and as a result, automatically update `i`.

Reply:
Done!